File: mobile/androidApp/src/main/kotlin/com/mentora/android/ui/components/SuccessState.kt
package com.mentora.android.ui.components

import android.provider.Settings
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.style.TextAlign
import com.mentora.android.theme.MentoraDimens
import com.mentora.android.theme.MentoraMotionDuration
import com.mentora.android.theme.MentoraMotionEasing
import com.mentora.android.theme.extendedColors

/**
 * Whether the system "remove animations" preference is active, checked via
 * `Settings.Global.ANIMATOR_DURATION_SCALE` (`0` when the user has that accessibility setting on).
 *
 * **Corrected disclosure — Compose's own animations already respect this setting.** An earlier version
 * of this kdoc claimed Compose's `animateFloatAsState`/`tween` do *not* consult
 * `Settings.Global.ANIMATOR_DURATION_SCALE`, and that only the View system's `ValueAnimator` does. That
 * was factually wrong, verified directly: Compose's `Recomposer`/frame clock implements
 * `MotionDurationScale`, which *does* read that system setting automatically — setting
 * `animator_duration_scale=0` causes a real `tween(300)` animation to complete in ~2 frames instead of
 * ~300ms, with zero manual code. So this explicit check is **redundant** with what Compose already does
 * on its own, not a gap Compose leaves open.
 *
 * **Why it's kept anyway.** [SuccessState]'s spec (`ACCESSIBILITY.md` § 9) additionally wants a
 * *different* animation shape under reduced motion — an opacity-only cross-fade instead of the normal
 * scale+fade, at a shorter duration — not just "the same animation, sped up," which is all
 * `MotionDurationScale` gives you automatically. This manual check drives that shape swap; it's
 * redundant with, not a replacement for, Compose's own automatic duration-scale handling, and is left in
 * place rather than removed since dropping it would regress that shape swap.
 *
 * **Check-once, by design, not a bug.** This reads the setting once via a keyless `remember` and never
 * re-observes a mid-session toggle. That's an intentional, disclosed tradeoff for this component
 * specifically (not a general pattern to copy): a success screen is a one-shot celebratory moment that
 * gets freshly recomposed from scratch each time it's shown (e.g. after a checkout completes), so
 * "checked once at composition start" already covers its real use case — there's no long-lived
 * [SuccessState] instance sitting on screen for a settings toggle to invalidate mid-display.
 */
@Composable
internal fun rememberReducedMotionEnabled(): Boolean {
    val context = LocalContext.current
    return remember {
        Settings.Global.getFloat(context.contentResolver, Settings.Global.ANIMATOR_DURATION_SCALE, 1f) == 0f
    }
}

/**
 * `design-system/COMPONENTS.md` § SuccessState (lines 568-583) — "structurally identical [to
 * EmptyState/ErrorState] ... but using the success semantic identity." Icon (`icon.large`,
 * `color.success.default`) -> title (`heading.h3`) -> description (`body.small`, max ~2 lines) ->
 * `PrimaryButton`, `space.10` vertical container padding.
 *
 * **Entrance motion + reduced motion (unconditional per the spec's own wording — "no exception for
 * this component just because it's a celebratory moment").** Normally: scale 0.9->1 + fade 0->1 over
 * `motion.duration.slow` + `easing.decelerate`. With reduced motion on ([rememberReducedMotionEnabled]):
 * an opacity-only cross-fade at `motion.duration.fast` — no scale — per `ACCESSIBILITY.md` § 9's
 * "cross-fade or cut instantly instead of playing slide/scale transitions."
 */
@Composable
fun SuccessState(
    title: String,
    description: String,
    actionLabel: String,
    onActionClick: () -> Unit,
    modifier: Modifier = Modifier,
    icon: MentoraIconName = MentoraIconName.CheckCircle,
) {
    val reducedMotion = rememberReducedMotionEnabled()
    var visible by remember { mutableStateOf(false) }
    LaunchedEffect(Unit) { visible = true }

    val durationMillis = if (reducedMotion) MentoraMotionDuration.fast else MentoraMotionDuration.slow
    val alpha by animateFloatAsState(
        targetValue = if (visible) 1f else 0f,
        animationSpec = tween(durationMillis = durationMillis, easing = MentoraMotionEasing.decelerate),
        label = "success-state-alpha",
    )
    val scale by animateFloatAsState(
        // Reduced motion drops the scale entirely (target pinned to 1f throughout) — only the alpha
        // cross-fade above plays, per this file's kdoc.
        targetValue = if (reducedMotion) 1f else if (visible) 1f else 0.9f,
        animationSpec = tween(durationMillis = durationMillis, easing = MentoraMotionEasing.decelerate),
        label = "success-state-scale",
    )

    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = MentoraDimens.spacing.space10)
            .graphicsLayer {
                this.alpha = alpha
                scaleX = scale
                scaleY = scale
            },
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(MentoraDimens.spacing.space3),
    ) {
        MentoraIcon(
            name = icon,
            contentDescription = null,
            size = MentoraDimens.iconSize.large,
            tint = MaterialTheme.extendedColors.success,
        )
        Text(
            text = title,
            style = MaterialTheme.typography.headlineSmall, // heading.h3.
            color = MaterialTheme.colorScheme.onSurface,
            textAlign = TextAlign.Center,
        )
        Text(
            // No maxLines/ellipsis clamp — CONTENT_RESILIENCE.md § 1 / COMPONENTS.md line 583's "~2
            // lines" is soft guidance for typical content length, not a hard truncation rule; this
            // wraps freely, matching ErrorState.kt's already-correct unclamped treatment.
            text = description,
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
        )
        PrimaryButton(text = actionLabel, onClick = onActionClick)
    }
}

File: mobile/iosApp/iosApp/Components/SuccessState.swift
import SwiftUI
import shared

// Phase 5 Task T11 slice 4 (Component Kit B) -- `design-system/COMPONENTS.md § SuccessState` (lines
// 568-583) -- "structurally identical [to EmptyState/ErrorState] ... but using the success semantic
// identity." Icon (`icon.large`, `color.success.default`) -> title (`heading.h3`) -> description
// (`body.small`) -> `PrimaryButton`, `space.10` vertical container padding. Ported directly from
// Android's own `SuccessState.kt`.
//
// **Entrance motion + reduced motion (unconditional per the spec's own wording).** Normally: scale
// 0.9->1 + fade 0->1 over `motion.duration.slow` + `easing.decelerate`. With Reduce Motion on: an
// opacity-only cross-fade at `motion.duration.fast` -- no scale -- per `ACCESSIBILITY.md § 9`'s
// "cross-fade or cut instantly instead of playing slide/scale transitions."
//
// **Simplification vs. Android, idiomatic-SwiftUI, not a gap.** Android's own `rememberReducedMotionEnabled()`
// manually reads `Settings.Global.ANIMATOR_DURATION_SCALE` because Compose's own kdoc there discloses
// that check is actually REDUNDANT with what Compose's animation system already does automatically --
// kept there only to drive a DIFFERENT animation shape (cross-fade vs. scale+fade), not because Compose
// needs help detecting the setting. `@Environment(\.accessibilityReduceMotion)` is this platform's own
// real, native, always-current equivalent signal (unlike Android's own kdoc-disclosed "checked once,
// never re-observed" limitation, this environment value is live and would update immediately if toggled
// while this view is on screen) -- reading it directly here needs no bespoke settings-polling code at
// all, so this port uses the idiomatic native mechanism rather than replicating Android's own
// platform-specific workaround line-for-line.
struct SuccessState: View {
    let title: String
    let description: String
    let actionLabel: String
    let onActionClick: () -> Void
    var icon: MentoraIconName = .checkCircle

    @Environment(\.accessibilityReduceMotion) private var reduceMotion
    @State private var isVisible = false

    private var duration: Double {
        reduceMotion ? MentoraMotionDuration.fast : MentoraMotionDuration.slow
    }

    var body: some View {
        VStack(spacing: MentoraSpacing.space3) {
            MentoraIcon(name: icon, size: MentoraIconSize.large)
                .foregroundStyle(Color.mentoraSuccessDefault)
                .accessibilityHidden(true)
            Text(title)
                .mentoraFont(.h3)
                .foregroundStyle(Color.mentoraTextPrimary)
                .multilineTextAlignment(.center)
            Text(description)
                .mentoraFont(.bodySmall)
                .foregroundStyle(Color.mentoraTextSecondary)
                .multilineTextAlignment(.center)
            PrimaryButton(label: actionLabel, action: onActionClick)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, MentoraSpacing.space10)
        .opacity(isVisible ? 1 : 0)
        .scaleEffect(reduceMotion ? 1 : (isVisible ? 1 : 0.9))
        .animation(MentoraMotionEasing.animation(MentoraMotionEasing.decelerate, duration: duration), value: isVisible)
        .onAppear { isVisible = true }
    }
}
